notebooks/visualization/matplotlib-intermediate.ipynb

```diff
@@ -0,0 +1,343 @@
+    "\n",
+    "\n",
+    "### Focuses\n",
+    "\n",
+    "* Draw multiple plots in the same figure.\n",
+    "* Colorize individual scatter plot points based on some criteria.\n",
+    "* Use imshow, contour and contourf to draw a heat maps and contour plots.\n",
+    "\n",
+    "\n",
+    "### Objectives\n",
+    "\n",
+    "1. [Create Test Data](#1.-Create-Test-Data)\n",
+    "1. [Multiple Plots on One Figure](#2.-Multiple-Plots-on-One-Figure)\n",
+    "1. [Colorizing Scatter Plots](#3.-Colorizing-Scatter-Plots)\n",
+    "1. [Plot a Heat Map with Imshow](#4.-Plot-a-Heat-Map-with-Imshow)\n",
+    "1. [Plot with Contour](#5.-Plot-with-Contour)\n",
+    "1. [Plot with Contourf](#6.-Plot-with-Contourf)\n",
+    "1. [Combine Imshow and Contour](#7.-Combine-Imshow-and-Contour)\n",
+    "\n",
+    "---"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Create Test Data\n",
+    "\n",
+    "First, instantiate some test data to work with in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create the test data to work with\n",
+    "times = np.array([ 93.,  96.,  99., 102., 105., 108., 111., 114., 117.,\n",
+    "                  120., 123., 126., 129., 132., 135., 138., 141., 144.,\n",
+    "                  147., 150., 153., 156., 159., 162.])\n",
+    "temps = np.array([310.7, 308.0, 296.4, 289.5, 288.5, 287.1, 301.1, 308.3,\n",
+    "                  311.5, 305.1, 295.6, 292.4, 290.4, 289.1, 299.4, 307.9,\n",
+    "                  316.6, 293.9, 291.2, 289.8, 287.1, 285.8, 303.3, 310.])\n",
+    "temps_1000 = np.array([316.0, 316.3, 308.9, 304.0, 302.0, 300.8, 306.2, 309.8,\n",
+    "                       313.5, 313.3, 308.3, 304.9, 301.0, 299.2, 302.6, 309.0,\n",
+    "                       311.8, 304.7, 304.6, 301.8, 300.6, 299.9, 306.3, 311.3])\n",
+    "# Fake dewpoint data to plot\n",
+    "dewpoint = 0.9 * temps\n",
+    "dewpoint_1000 = 0.9 * temps_1000"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<a href=\"#top\">Top</a>\n",
+    "\n",
+    "---\n",
+    "\n",
+    "## 2. Multiple Plots on One Figure\n",
+    "\n",
+    "The [Matlibplot: Basics]() notebook explained how to create a figure and add a plot to that figure.  It is also possible to add multiple plots to the same figure.  In this step, a figure is first created and then multiple plots are added, each with their own titles, axes, etc."
```

Comment:
I'd add an explanation about how to add multiple plots, like:
"This is accomplished by calling `fig.add_subplot()`. This function lays out multiple plots in a grid, by passing in the number of rows, number of columns, and the number of the plot within the grid. So for example, if we wanted to create a plot in the lower left corner of a (hypothetical) grid of plots with 2 rows and 3 columns, we could write `fig.add_subplot(2, 3, 4)`, where 4 means the 4th plot, counting across rows and down columns."

Reply:
Added this change with updated PR